admin/action/modules/fman.php

```diff
@@ -320,7 +322,43 @@
                 }
                 break;
 
-                // odstraneni vyberu
+            // stazeni souboru
+            case "downloadselected":
+                // ziskani vybranych souboru
+                $selected = array();
+                foreach ($_POST as $var => $val) {
+                    if ($var == "action" || $var == "param") {
+                        continue;
+                    }
+                    $val = $decodeFilename($val);
+                    if (file_exists($dir . $val) && !is_dir($dir . $val) && User::checkFilename($val)) {
+                        $selected[$val] = $dir.$val;
```

Comment:
I'd do `$selected[] = $val` instead. The directory can be added when calling `$zip->addFile()`.

Reply:
You're right, that's better.

Reply:
Done